### ISSUE TITLE
CentOS detection is shaky

### ISSUE THREAD

**Issue by hasufell** [haskell/ghcup#32]
On CentOS 6.7 we get:

```
Detected system information:
  Architecture:   x86_64
  Distribution:   Linux
  Distro version: 4.17.14
```

The only file with useful information seems to be:

```
# cat /etc/centos-release
CentOS release 6.7 (Final)
```

**Comment by hvr** [comment]
Another issue I ran into is that `uname -m` is utterly unreliable in discriminating i386 vs x86_64 userlands; instead, on Linux, `getconf LONG_BIT` should be used which will output either `32` or `64` depending on whether we have a 32bit or 64bit userland.

**Comment by hvr** [comment]
I believe the major pain points of this ticket to have been addressed via #55

The remaining issue is of handling a 32bit-userland w/ a 64bit-kernel is not CentOS specific

**Comment by hasufell** [comment]
> The remaining issue is of handling a 32bit-userland w/ a 64bit-kernel is not CentOS specific

Mind to create a PR?
